### PR TITLE
Transpose square matrix

### PR DESCRIPTION
**Pull request by ANSHIKA1806**
It is an important question and it should be in the project.
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes
(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [ yes] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#4867

### CODE REVIEW

File: Java/other/Transpose square matrix.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+public class MatrixTransposeExample{  
+public static void main(String args[]){  
+//creating a matrix  
+int original[][]={{1,3,4},{2,4,3},{3,4,5}};
```
Comment (siddharth25pandey) [inline Java/other/Transpose square matrix.java]: Take Dynamic Input

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ANSHIKA1806 Do the mentioned changes & add the update in **README FILE**

**Comment by ANSHIKA1806** [comment]
@siddharth25pandey I've made the changes. Please check it and merge it.

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey I've made the changes. Please check it and merge it.

Please read complete [message](https://github.com/TesseractCoding/NeoAlgo/pull/5108#pullrequestreview-630180386)

**Comment by siddharth25pandey** [comment]
@ANSHIKA1806 Also, include Sample Input & Output and Time & Space Complexity at the end of Program in commented form.

File: Java/other/Transpose square matrix.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import java.util.Scanner;  
+public class MatrixTransposeExample2
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java]: Rewrite it as `MatrixTranspose`

File: Java/other/Transpose square matrix.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }  
+    System.out.println("The above matrix before Transpose is ");  
+    for(i = 0; i < row; i++)  
+        {  
+            for(j = 0; j < column; j++)  
+            {  
+            System.out.print(array[i][j]+" ");  
+            }  
+            System.out.println(" ");  
+        }
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java]: Align your braces properly.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

File: Java/other/Transpose square matrix.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    int column = s.nextInt();  
+    int array[][] = new int[row][column];  
+    System.out.println("Enter matrix:");  
+    for(i = 0; i < row; i++)  
+    {  
+        for(j = 0; j < column; j++)   
+            {  
+            array[i][j] = s.nextInt();  
+            System.out.print(" ");  
+            }
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java]: Add a one tab space inside the body

File: Java/other/Transpose square matrix.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            System.out.print(" ");  
+            }  
+    }  
+    System.out.println("The above matrix before Transpose is ");  
+    for(i = 0; i < row; i++)  
+        {  
+            for(j = 0; j < column; j++)  
+            {  
+            System.out.print(array[i][j]+" ");  
+            }
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java]: Add a one tab space inside the body

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by ANSHIKA1806** [comment]
@ankitaggarwal23 @siddharth25pandey I've made the changes, Please check it and merge it.

File: Java/other/Transpose square matrix.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+//Transpose of a matrix is obtained by changing rows to columns and columns to rows. In other words, transpose of A[][] is obtained by changing A[i][j] to A[j][i].
+
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:2]: - Use multiline comments
- Make it short and understandable
- Add a bit more description

File: Java/other/Transpose square matrix.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        int row = s.nextInt(); 
+        int column = s.nextInt();
+        int array[][] = new int[row][column];
+        System.out.println("Enter matrix:");
+        for(i = 0; i < row; i++)
+        {
+            for(j = 0; j < column; j++) 
+                {
+                    array[i][j] = s.nextInt(); // for taking input
+                    System.out.print(" "); //new line
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:20]: Remove redundant comments

File: Java/other/Transpose square matrix.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                }
+        }
+        System.out.println("The above matrix before Transpose is ");
+        for(i = 0; i < row; i++)
+        {
+            for(j = 0; j < column; j++)
+            {
+                System.out.print(array[i][j]+" "); //Priting matrix as entered
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:28]: Remove redundant comments

File: Java/other/Transpose square matrix.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            System.out.println(" ");
+        }
+        System.out.println("The above matrix after Transpose is ");
+        for(i = 0; i < column; i++)
+        {
+            for(j = 0; j < row; j++)
+            {
+                System.out.print(array[j][i]+" "); //Printing transposed matrix
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:37]: Remove redundant comments

File: Java/other/Transpose square matrix.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        int column = s.nextInt();
+        int array[][] = new int[row][column];
+        System.out.println("Enter matrix:");
+        for(i = 0; i < row; i++)
+        {
+            for(j = 0; j < column; j++) 
+                {
+                    array[i][j] = s.nextInt(); // for taking input
+                    System.out.print(" "); //new line
+                }
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:21]: Align it properly braces are not aligned.

File: Java/other/Transpose square matrix.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        System.out.println("The above matrix after Transpose is ");
+        for(i = 0; i < column; i++)
+        {
+            for(j = 0; j < row; j++)
+            {
+                System.out.print(array[j][i]+" "); //Printing transposed matrix
+            }
+            System.out.println(" ");
+            }
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:40]: Remove extra braces. As code produces error.

File: Java/other/Transpose square matrix.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+9
+The above matrix before Transpose is 
+1 2 3  
+4 5 6  
+7 8 9  
+The above matrix after Transpose is 
+1 4 7  
+2 5 8  
+3 6 9
+*/
```
Comment (ankitaggarwal23) [inline Java/other/Transpose square matrix.java:65]: Rewrite it as,

```
/*
Sample Output:

Enter total rows and columns: 
3 3
Enter matrix:
1
2
3
4
5
6
7
8
9
The above matrix before Transpose is 
1 2 3  
4 5 6  
7 8 9  
The above matrix after Transpose is 
1 4 7  
2 5 8  
3 6 9

Time Complexity: 
Space Complexity: 
*/

```

Add TC and SC

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity. Thank you for your valuable time.😊

**Pull request closed without merging**